### PR TITLE
Feat: Add distributor creation block to fetch list

### LINKED ISSUE #149: Feat: Add distributor creation block to fetch list
## Context

Each distributor needs its balance captured at the exact time of creation. This provides the initial balance state and ensures complete balance history from distributor inception.

## Current Behavior

1. Only end-of-day blocks are considered for balance fetching
2. Distributor creation block is not included in the fetch list
3. Initial balance at creation time is missing from historical data

## Expected Behavior

- Parse creation block number from distributor metadata (if available)
- If creation block exists, add it to the filtered date/block map
- Use creation date as the key if block not already present for that date
- Handle case where creation date already has an end-of-day block (don't duplicate)
- Ensure creation block is processed before other dates (chronological order)

## Required Context

- `src/types/index.ts` - DistributorInfo metadata structure
- `__tests__/test-data/distributor-detector/distributor-creation-events-raw.json` - example creation data
- `docs/specs/balance-fetcher.md` - requirement to fetch creation time balance

## Out of Scope

- Loading existing balance data
- Identifying missing balances
- Making RPC calls
- Balance data storage

### PR DESCRIPTION
**Pull request by yahgwai**
## Summary
- Add distributor creation blocks to the balance fetch list to capture initial balance state
- Ensure creation time balance is fetched in addition to end-of-day balances
- Handle edge cases for distributors created mid-day vs those with future creation dates

## Implementation Details
Following TDD approach:
1. **RED**: Added failing tests for the new functionality
2. **GREEN**: Implemented minimal code to make tests pass
3. **REFACTOR**: Improved code clarity and maintainability

### Changes:
- Modified `BalanceFetcher.fetchBalances()` to include creation blocks when needed
- Creation block is added only when its date doesn't already have an end-of-day block
- All blocks are fetched in chronological order across multiple distributors
- Future distributors (with no applicable blocks) are skipped

## Test plan
- [x] Unit tests added for all new functionality
- [x] All existing tests still pass
- [x] TypeScript compilation successful
- [x] ESLint checks pass
- [x] Code properly formatted with Prettier

Closes #149

### CODE REVIEW

File: src/balance-fetcher.ts [hunk 1]
```diff
@@ -54,21 +54,49 @@ export class BalanceFetcher {
...
       }
+
+      // Collect all blocks for this distributor
+      for (const [date, block] of endOfDayBlocks) {
+        allFetches.push({ address, date, block });
+      }
+    }
+
+    // Sort all fetches chronologically by date
+    allFetches.sort((a, b) => a.date.localeCompare(b.date));
```
Comment (yahgwai) [inline src/balance-fetcher.ts:95]: Is localCompare the right thing to do here? what kind of dates are these? It might be, I just dont know

**Comment by yahgwai** [comment]
Good question\! Yes, `localeCompare` is the right choice here. 

The dates are in ISO 8601 format (YYYY-MM-DD), which are lexicographically sortable. Both `localeCompare` and regular string comparison would work correctly for this format:

- `"2022-08-10" < "2023-03-16"` ✓
- `"2022-08-10".localeCompare("2023-03-16") < 0` ✓

I chose `localeCompare` because:
1. It's more explicit about comparing strings as locale-aware values
2. It's consistent with best practices for string comparison
3. It returns the standard -1/0/1 pattern that sort expects

I've added a test in commit fa6be27 that demonstrates both approaches work identically for ISO date strings.
